From: NeilBrown <neilb@suse.com>
Subject: Fix kabi after "md: batch flush requests."
Patch-mainline: Never, kabi
References: bsc#1119680

The new fields are private to md, so they can be hidden from the kabi.

Acked-by: NeilBrown <neilb@suse.com>
Signed-off-by: Neil Brown <neilb@suse.com>
Acked-by: Coly Li <colyli@suse.de>
---
 drivers/md/md-cluster.h |    5 ++++-
 drivers/md/md.h         |   20 +++++++++++++++-----
 2 files changed, 19 insertions(+), 6 deletions(-)

--- a/drivers/md/md-cluster.h
+++ b/drivers/md/md-cluster.h
@@ -25,10 +25,13 @@ struct md_cluster_operations {
 	int (*remove_disk)(struct mddev *mddev, struct md_rdev *rdev);
 	void (*load_bitmaps)(struct mddev *mddev, int total_slots);
 	int (*gather_bitmaps)(struct md_rdev *rdev);
-	int (*resize_bitmaps)(struct mddev *mddev, sector_t newsize, sector_t oldsize);
 	int (*lock_all_bitmaps)(struct mddev *mddev);
 	void (*unlock_all_bitmaps)(struct mddev *mddev);
 	void (*update_size)(struct mddev *mddev, sector_t old_dev_sectors);
+
+#ifndef __GENKSYMS__
+	int (*resize_bitmaps)(struct mddev *mddev, sector_t newsize, sector_t oldsize);
+#endif
 };
 
 #endif /* _MD_CLUSTER_H */
--- a/drivers/md/md.h
+++ b/drivers/md/md.h
@@ -463,22 +463,29 @@ struct mddev {
 						   * metadata and bitmap writes
 						   */
 
+#ifdef __GENKSYMS__
+	mempool_t                       *flush_pool;
+	mempool_t                       *flush_bio_pool;
+#else
 	/* Generic flush handling.
 	 * The last to finish preflush schedules a worker to submit
 	 * the rest of the request (without the REQ_PREFLUSH flag).
 	 */
 	struct bio *flush_bio;
 	atomic_t flush_pending;
-	ktime_t start_flush, last_flush; /* last_flush is when the last completed
-					  * flush was started.
-					  */
-	struct work_struct flush_work;
+#endif
 	struct work_struct event_work;	/* used by dm to report failure event */
 	void (*sync_super)(struct mddev *mddev, struct md_rdev *rdev);
 	struct md_cluster_info		*cluster_info;
 	unsigned int			good_device_nr;	/* good device num within cluster raid */
 
 	bool	has_superblocks:1;
+#ifndef __GENKSYMS__
+	struct work_struct flush_work;
+	ktime_t start_flush, last_flush; /* last_flush is when the last completed
+					  * flush was started.
+					  */
+#endif
 };
 
 enum recovery_flags {
@@ -558,7 +565,6 @@ struct md_personality
 	int (*check_reshape) (struct mddev *mddev);
 	int (*start_reshape) (struct mddev *mddev);
 	void (*finish_reshape) (struct mddev *mddev);
-	void (*update_reshape_pos) (struct mddev *mddev);
 	/* quiesce suspends or resumes internal processing.
 	 * 1 - stop new actions and wait for action io to complete
 	 * 0 - return to normal behaviour
@@ -579,6 +585,10 @@ struct md_personality
 	int (*congested)(struct mddev *mddev, int bits);
 	/* Changes the consistency policy of an active array. */
 	int (*change_consistency_policy)(struct mddev *mddev, const char *buf);
+
+#ifndef __GENKSYMS__
+	void (*update_reshape_pos) (struct mddev *mddev);
+#endif
 };
 
 struct md_sysfs_entry {
